fermipy/gtanalysis.py

```diff
@@ -4024,54 +4024,62 @@ def get_src_model(self, name, paramsonly=False, reoptimize=False,
             thesrc = self.like[name]
             src_dict['flux'] = self.like.flux(name, self.energies[0],
                                               self.energies[-1])
-            src_dict['flux100'] = self.like.flux(name, 100., 10 ** 5.5)
-            src_dict['flux1000'] = self.like.flux(name, 1000., 10 ** 5.5)
-            src_dict['flux10000'] = self.like.flux(name, 10000., 10 ** 5.5)
             src_dict['eflux'] = self.like.energyFlux(name,
                                                      self.energies[0],
                                                      self.energies[-1])
-            src_dict['eflux100'] = self.like.energyFlux(name, 100.,
-                                                        10 ** 5.5)
-            src_dict['eflux1000'] = self.like.energyFlux(name, 1000.,
-                                                         10 ** 5.5)
-            src_dict['eflux10000'] = self.like.energyFlux(name, 10000.,
-                                                          10 ** 5.5)
-            src_dict['dnde'] = self.like[name].spectrum()(
-                pyLike.dArg(src_dict['pivot_energy']))
-            src_dict['dnde100'] = self.like[name].spectrum()(
-                pyLike.dArg(100.))
-            src_dict['dnde1000'] = self.like[name].spectrum()(
-                pyLike.dArg(1000.))
-            src_dict['dnde10000'] = self.like[name].spectrum()(
-                pyLike.dArg(10000.))
+            emax = np.min([self.energies.max(), 10 ** 5.5])
+            for e in np.logspace(2.,4.,3):
```

Comment:
Since this `np.logspace` does not contain any variable, it would be better to explicitly mention the energies, like:

```python
for e in [100, 1000, 10000]:
...
```

Also you use this almost exact code multiple times, it would be cleaner to put it in a function.